src/aggregation_control.rs

```diff
@@ -54,3 +54,130 @@ impl AggregationController {
+        let mut files = Vec::new();
+        for _ in 0..num_files {
+            files.push(test_file_path_buf.clone())
+        }
+        let mut file_handler_receivers = Vec::new();
+        let mut file_handler_senders = Vec::new();
+        let (agg_sndr, agg_recv) = mpsc::channel();
+        for _ in 0..num_file_handlers {
+            let (sndr, recv) = mpsc::channel();
+            file_handler_receivers.push(recv);
+            file_handler_senders.push(sndr);
+        }
+        for _ in 0..num_files + 1 {
+            let _ = agg_sndr.send(AggregationMessages::Next(file_handler_of_interest));
+        }
+        // Once all of the file_handlers are finished the main loop of the controller will shutdown.
+        for _ in 0..num_file_handlers {
+            let _ = agg_sndr.send(AggregationMessages::Aggregate(test_common::TEST_LOG_FILE_AGGS,
+                                                                 HashMap::new()));
+        }
+        let mut agg_ctrl = super::AggregationController::new(agg_recv, file_handler_senders);
+        agg_ctrl.run_aggregation(&mut files);
+        for _ in 0..num_files {
+            let received_msg = file_handler_receivers[file_handler_of_interest].recv();
+
+            assert_eq!(received_msg.unwrap(),
+                       FileHandlingMessages::Filename(test_file_path_buf.clone()));
+        }
+        let expected_done_msg = file_handler_receivers[file_handler_of_interest].recv();
+
+        assert_eq!(expected_done_msg.unwrap(), FileHandlingMessages::Done);
+    }
+
+    #[test]
+    fn run_aggregation_sends_a_filename_to_the_correct_file_handler_when_there_is_a_filename() {
+        let num_file_handlers = 10;
+        let file_handler_of_interest = 3;
+        let test_file_path_buf = PathBuf::from(test_common::TEST_LOG_FILE);
+        let mut files = vec![test_file_path_buf.clone()];
+        let mut file_handler_receivers = Vec::new();
+        let mut file_handler_senders = Vec::new();
+        let (agg_sndr, agg_recv) = mpsc::channel();
+        for _ in 0..num_file_handlers {
+            let (sndr, recv) = mpsc::channel();
+            file_handler_receivers.push(recv);
+            file_handler_senders.push(sndr);
+        }
+
+        let _ = agg_sndr.send(AggregationMessages::Next(file_handler_of_interest));
+        // Once all of the file_handlers are finished the main loop of the controller will shutdown.
+        for _ in 0..num_file_handlers {
+            let _ = agg_sndr.send(AggregationMessages::Aggregate(test_common::TEST_LOG_FILE_AGGS,
+                                                                 HashMap::new()));
+        }
+        let mut agg_ctrl = super::AggregationController::new(agg_recv, file_handler_senders);
+        agg_ctrl.run_aggregation(&mut files);
+        let received_msg = file_handler_receivers[file_handler_of_interest].recv();
+
+        assert_eq!(received_msg.unwrap(),
```

Comment:
Hmmmm... There may be a way using recv_timeout but that test would not be correct because it ensures the other channels didn't receive a message before the timeout only. That's doesn't mean a message wasn't sent after the timeout (in a long running application).

I don't think that would add much value as a test? What do you think?